packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -47,6 +47,7 @@ class FlutterDevice {
     TargetModel targetModel = TargetModel.flutter,
     TargetPlatform targetPlatform,
     ResidentCompiler generator,
+    this.userIdentifier,
```

Comment:
I started by adding `userIdentifier` to `DebuggingOptions` property bag which meant that `userIdentifier` was explicitly passed around in fewer places.  However that didn't really make sense--the user isn't relevant for debugging or build, it's only relevant once you go to install or start the prebuilt app.  So I moved it into FlutterDevice.  You could theoretically deploy to multiple users on the same device, but I don't picture needing to happening often so it's not supported.

I'm all ears if you guys have a better suggestion for how to inject this into the AndroidDevice install/uninstall/start methods.

Reply:
I agree that it doesn't really make sense for debuggingOptions. Updating the Device APIs seems like a reasonable approach - though I'm fairly certain there are overrides in google3, so you might want to double check even if FROB passed

Reply:
> though I'm fairly certain there are overrides in google3, so you might want to double check even if FROB passed

I don't see any obvious overrides that would be impacted by this change.  @mehmetf can you double check me?  You were thinking about this awhile ago
https://yaqs.corp.google.com/eng/q/4970372142465024 (internal)
b/142944798

Reply:
Never mind, @jonahwilliams pointed me in the right direction.

Reply:
We override the DeviceManager. If tests pass, this should be OK to land. Feel free to update the internal bug so we can try it out internally when this lands.

Reply:
Supporting CL has been submitted.